nilearn/signal.py

```diff
@@ -641,3 +562,129 @@ def clean(signals, sessions=None, detrend=True, standardize='zscore',
                                detrend=False)
 
     return signals
+
+
+def _filter_signal(signals, confounds, filter, low_pass, high_pass, t_r):
+    '''Filter signal based on provided strategy.'''
+    if filter == 'butterworth':
+        signals = butterworth(signals, sampling_rate=1. / t_r,
+                              low_pass=low_pass, high_pass=high_pass)
+        if confounds is not None:
+            # Apply low- and high-pass filters to keep filters orthogonal
+            # (according to Lindquist et al. (2018))
+            confounds = butterworth(confounds, sampling_rate=1. / t_r,
+                                    low_pass=low_pass, high_pass=high_pass)
+    elif filter == "cosine":
+        from .glm.first_level.design_matrix import _cosine_drift
+        frame_times = np.arange(signals.shape[0]) * t_r
+        cosine_drift = _cosine_drift(high_pass, frame_times)
+        confounds = np.hstack((confounds, cosine_drift))
```

Comment:
The last column of output from _cosine_drift will always be a column of ones, but it doesn't hurt to do a check here.

Reply:
Would there be an issue with the bias term of `_cosine_drift` being redundant with `standardize=True`? Since the mean will be removed anyway? Or is this a non-problem?